featuretools/computational_backends/calculate_feature_matrix.py

```diff
@@ -751,3 +751,20 @@ def update_progress_callback_parameters(progress_bar, previous_progress):
     progress_percent = (progress_bar.n / progress_bar.total) * 100
     time_elapsed = progress_bar.format_dict["elapsed"]
     return (update, progress_percent, time_elapsed)
+
+
+def init_ww_and_concat_fm(feature_matrix, ww_init_kwargs):
+    for fm in feature_matrix:
+        fm.ww.init(**ww_init_kwargs)
+
+    fm = feature_matrix[0]
+
+    for other in feature_matrix[1:]:
+        fm = fm.ww.append(other)
+        if not fm.ww.schema:
+            # Koalas bug sometimes causes schema to be invalidated
+            # after `append` so need to reinitialize WW
+            fm.ww.init(**ww_init_kwargs)
```

Comment:
another option that might handle this case is to do the `append` call outside of Woodwork and then reinit with the schema. Then you wouldn't get the `TypingInfoMismatchWarning`.

```python
schema = fm.ww.schema
fm = fm.append(other) #or can use pd.concat now that you don't need to go through woodwork accessor
fm.ww.init(schema=schema)
```
This will make it so that the init call can update any dtypes or underlying index (since it'll use the partial schema init unlike the `fm.ww.append` call which goes through the full schema init, which requires the schema be valid). The downside would be if you wanted to be able to catch that `not fm.ww.schema` case so that you can re-init in the exact way you are now, you'd lose that control.

Reply:
Though, this is also a point in the column of adding `ww.concat_rows` or updating `ww.concat_cols` to be a more general concat.